akka-docs/rst/scala/code/docs/akka/typed/IntroSpec.scala

```diff
@@ -0,0 +1,206 @@
+/**
+ * Copyright (C) 2014 Typesafe Inc. <http://www.typesafe.com>
+ */
+package docs.akka.typed
+
+//#imports
+import akka.typed._
+import akka.typed.ScalaDSL._
+import akka.typed.AskPattern._
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.concurrent.Await
+//#imports
+import akka.testkit.AkkaSpec
+import akka.typed.TypedSpec
+
+object IntroSpec {
+
+  //#hello-world-actor
+  object HelloWorld {
+    final case class Greet(whom: String, replyTo: ActorRef[Greeted])
+    final case class Greeted(whom: String)
+
+    val greeter = Static[Greet] { msg ⇒
+      println(s"Hello ${msg.whom}!")
+      msg.replyTo ! Greeted(msg.whom)
+    }
+  }
+  //#hello-world-actor
+
+  //#counter-actor
+  object Counter {
+
+    sealed trait Command
+    final case class GetAndAdd(amount: Int, replyTo: ActorRef[CurrentCount]) extends Command
+    final case class AddAndGet(amount: Int, replyTo: ActorRef[CurrentCount]) extends Command
+
+    final case class CurrentCount(x: Int)
+
+    val counter: Behavior[Command] = behavior(0)
```

Comment:
yes, it works for immutable and it would be great to clarify that in the docs if that is not already done.
I just don't want users to think that this is always the recommended way to do it, then we are back before akka 2.0. Users tend to copy without thinking.